Scripts/RemediationScripts/Remediate-EnableSSLDBForMySQLFlexibleServer.ps1

```diff
@@ -0,0 +1,804 @@
+                    $logResource = @{}	
+                    $logResource.Add("ResourceGroupName", ($_.ResourceGroupName))	
+                    $logResource.Add("ResourceName", ($_.ResourceName))	
+                    $logRemediatedResources += $logResource	
+                }
+                else {
+                    $DBForMySQLFSSkipped += $_
+                    $logResource = @{}	
+                    $logResource.Add("ResourceGroupName", ($_.ResourceGroupName))	
+                    $logResource.Add("ResourceName", ($_.ResourceName))
+                    $logResource.Add("Reason", "Encountered error while setting server parameter require_secure_transport: [$($DBForMySQLFS)]")            
+                    $logSkippedResources += $logResource	
+                }                
+            }
+            catch {
+                $DBForMySQLFSSkipped += $_
+                $logResource = @{}	
+                $logResource.Add("ResourceGroupName", ($_.ResourceGroupName))	
+                $logResource.Add("ResourceName", ($_.ResourceName))	
+                $logResource.Add("Reason", "Encountered error while setting server parameter require_secure_transport")    	
+                $logSkippedResources += $logResource	
+                Write-Host "Skipping this resource..." -ForegroundColor $([Constants]::MessageType.Warning)	
+                Write-Host $([Constants]::SingleDashLine)
+            }
+        }
+        Write-Host $([Constants]::DoubleDashLine)        
+        
+        if ($AutoRemediation) {
+            if ($($DBForMySQLFSRemediated | Measure-Object).Count -gt 0) {        
+                # Write this to a file.
+                $DBForMySQLFSRemediatedFile = "$($backupFolderPath)\RemediatedDBForMySQLFS.csv"
+                $DBForMySQLFSRemediated | Export-CSV -Path $DBForMySQLFSRemediatedFile -NoTypeInformation
+
+                Write-Host "The information related to Azure Database for MySQL flexible server(s) where server parameter require_secure_transport changed has been saved to [$($DBForMySQLFSRemediatedFile)]. Use this file for any roll back that may be required." -ForegroundColor $([Constants]::MessageType.Update)
+                Write-Host $([Constants]::SingleDashLine)
+            }        
+            if ($($DBForMySQLFSSkipped | Measure-Object).Count -gt 0) {
+                $DBForMySQLFSSkipped | Format-Table -Property $colsProperty -Wrap            
+                # Write this to a file.
+                $DBForMySQLFSSkippedFile = "$($backupFolderPath)\SkippedDBForMySQLFS.csv"
+                $DBForMySQLFSSkipped | Export-CSV -Path $DBForMySQLFSSkippedFile -NoTypeInformation
+                Write-Host "The information related to Azure Database for MySQL flexible server(s) where server parameter require_secure_transport not changed has been saved to [$($DBForMySQLFSSkippedFile)]. Use this file for any roll back that may be required." -ForegroundColor $([Constants]::MessageType.Warning)
```

Comment:
The statement, use this file for any rollback is not required as these are anyway skipped.